src/model/channel/guild_channel.rs

```diff
@@ -702,6 +702,60 @@ impl GuildChannel {
     #[cfg(feature = "http")]
     #[inline]
     pub fn webhooks(&self, http: impl AsRef<Http>) -> Result<Vec<Webhook>> { self.id.webhooks(&http) }
+
+    /// Retrieves [`Member`]s from the current channel.
+    ///
+    /// [`ChannelType::Voice`] returns [`Member`]s using the channel.
+    /// [`ChannelType::Text`] and [`ChannelType::News`] return [`Member`]s
+    /// that can read the channel.
+    ///
+    /// Other [`ChannelType`]s lack the concept of [`Member`]s and
+    /// will return: [`ModelError::InvalidChannelType`].
+    ///
+    /// [`Member`]: ../guild/struct.Member.html
+    /// [`ChannelType`]: enum.ChannelType.html
+    /// [`ChannelType::Voice`]: enum.ChannelType.html#variant.Voice
+    /// [`ChannelType::Text`]: enum.ChannelType.html#variant.Text
+    /// [`ChannelType::News`]: enum.ChannelType.html#variant.News
+    /// [`ModelError::InvalidChannelType`]: ../error/enum.Error.html#variant.InvalidChannelType
+    #[cfg(feature = "cache")]
+    #[inline]
+    pub fn members(&self, cache: impl AsRef<CacheRwLock>) -> Result<Vec<Member>> {
+           let cache = cache.as_ref();
+    let guild = cache.read().guild(self.guild_id).unwrap();
+
+    match self.kind {
+        ChannelType::Voice => Ok(guild
+            .read()
+            .voice_states
+            .values()
+            .filter_map(|v| {
+                v.channel_id.map_or_else(
```

Comment:
Unless it's expensive to compute the default value, using the `_else` variant is unnecessary here.
```suggestion
                v.channel_id.map_or(None, ...
```